### DISCUSSION TITLE
setValue inside useEffect not adding array field

### DISCUSSION THREAD

**Discussion by marr** [react-hook-form/react-hook-form#5820] [Q&A]
Hi, I have created a sandbox illustrating an issue I can't seem to crack.

The issue is that I get some array data via a state update, that I send to my form component via props, and I try to call setValue in a loop to set the values in the form. I'd expect the form to render all of the new fields, however only the first one is rendered.

You can see the behavior in this sandbox: https://codesandbox.io/s/gifted-clarke-r1vnv?file=/src/App.js. Just click "Add Data" and see that there is only one row set, even though I do a `forEach` over the data and call `setValue` on with the field name set to `users[index].name`.

**Comment by bluebill1049** [comment]
setValue doesn't really trigger re-render (unless watched), i think this method is what you after: https://github.com/react-hook-form/react-hook-form/discussions/5832
